hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentViewTest.java

```diff
@@ -0,0 +1,146 @@
+
+        // then
+        assertThat(deletions.getSubscriptions()).isEmpty();
+    }
+
+    @Test
+    public void shouldAddAssignmentToExistingSubscription() {
+        // given
+        SubscriptionName s1 = anySubscriptionName();
+        SubscriptionAssignmentView current = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1")));
+        SubscriptionAssignmentView target = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1"), assignment(s1, "c2")));
+
+        // when
+        SubscriptionAssignmentView additions = current.additions(target);
+
+        // then
+        assertThat(additions.getAssignmentsForSubscription(s1)).containsOnly(assignment(s1, "c2"));
+    }
+
+    @Test
+    public void shouldDeleteAssignmentFromExistingSubscription() {
+        // given
+        SubscriptionName s1 = anySubscriptionName();
+        SubscriptionAssignmentView current = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1"), assignment(s1, "c2")));
+        SubscriptionAssignmentView target = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1")));
+
+        // when
+        SubscriptionAssignmentView deletions = current.deletions(target);
+
+        // then
+        assertThat(deletions.getAssignmentsForSubscription(s1)).containsOnly(assignment(s1, "c2"));
+    }
+
+    @Test
+    public void shouldAddNewSubscription() {
+        // given
+        SubscriptionName s1 = anySubscriptionName();
+        SubscriptionName s2 = anySubscriptionName();
+        SubscriptionAssignmentView current = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1")));
+        SubscriptionAssignmentView target = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1")), s2, ImmutableSet.of(assignment(s2, "c1")));
+
+        // when
+        SubscriptionAssignmentView additions = current.additions(target);
+
+        // then
+        assertThat(additions.getSubscriptions()).containsOnly(s2);
+        assertThat(additions.getAssignmentsForSubscription(s2)).containsOnly(assignment(s2, "c1"));
+    }
+
+    @Test
+    public void shouldDeleteOldSubscription() {
+        // given
+        SubscriptionName s1 = anySubscriptionName();
+        SubscriptionAssignmentView current = subscriptionAssignmentView(s1, ImmutableSet.of(assignment(s1, "c1")));
+        SubscriptionAssignmentView target = subscriptionAssignmentView(s1, Collections.emptySet());
+
+        // when
+        SubscriptionAssignmentView deletions = current.deletions(target);
+
+        // then
+        assertThat(deletions.getSubscriptions()).isEqualTo(current.getSubscriptions());
+        assertThat(deletions.getAssignmentsForSubscription(s1)).isEqualTo(current.getAssignmentsForSubscription(s1));
+    }
+
+    private SubscriptionAssignmentView subscriptionAssignmentView(SubscriptionName s1, Set<SubscriptionAssignment> assignments) {
+        return new SubscriptionAssignmentView(ImmutableMap.of(s1, assignments));
```

Comment:
how about test subscriptionassignment builder? i think there are places cattered all over those tests where it could be used